go/arrow/cdata/cdata_exports.go

```diff
@@ -376,7 +381,7 @@ func exportArray(arr arrow.Array, out *CArrowArray, outSchema *CArrowSchema) {
 		// unions don't have validity bitmaps, but we keep them shifted
 		// to make processing easier in other contexts. This means that
 		// we have to adjust for union arrays
-		if arr.DataType().ID() == arrow.DENSE_UNION || arr.DataType().ID() == arrow.SPARSE_UNION {
+		if !internal.HasValidityBitmap(arr.DataType().ID(), flatbuf.MetadataVersionV5) {
```

Comment:
you can use `DefaultHasValidityBitmap` to avoid having to specify the metadataversion (it'll always default to the most recent)